static/app/views/dashboardsV2/detail.tsx

```diff
@@ -196,6 +200,10 @@ class DashboardDetail extends Component<Props, State> {
               source: DashboardWidgetSource.DASHBOARDS,
             });
           },
+          disableEditWidget:
+            organization.features.includes('dashboards-top-level-filter') &&
+            hasUnsavedFilterChanges(dashboard, location, dashboard.filters),
```

Comment:
this doesn't capture changes to the `filters` dict since we're just passing `dashboard.filters` and comparing it to the filters in `dashboard` so the edit widget button isn't disabled when there's an unsaved state with releases

Reply:
Fixed! Added `modifiedDashboard ?? dashboard` so we get the most up to date version of `filters`